openbook_auth/models.py

```diff
@@ -1909,6 +1918,22 @@ def _check_username_not_taken(self, username):
                 _('The username is already taken.')
             )
 
+    def _check_can_edit_comment_with_id_for_post_with_id(self, post_comment_id, post_id):
+        # Check that the comment belongs to the post
+        PostComment = get_post_comment_model()
+        Post = get_post_model()
+
+        if not PostComment.objects.filter(id=post_comment_id, post_id=post_id).exists():
+            raise ValidationError(
+                _('The comment does not belong to the specified post.')
+            )
+
+        if not self.posts_comments.filter(id=post_comment_id).exists():
```

Comment:
We can make this into a `has_post_comment_with_id` function and then use it in a `_check_has_post_comment_with_id` function which throws this exception and we just call here.